packages/nimble-components/src/rich-text-editor/tests/rich-text-editor.spec.ts

```diff
@@ -447,4 +447,662 @@ describe('RichTextEditor', () => {
+            const specType = getSpecTypeByNamedList(value, focused, disabled);
+            // eslint-disable-next-line @typescript-eslint/no-loop-func
+            specType(
+                `markdown string "${value.name}" returns as plain text "${value.name}" without any change`,
+                // eslint-disable-next-line @typescript-eslint/no-loop-func
+                async () => {
+                    element.setMarkdown(value.name);
+
+                    await connect();
+
+                    expect(element.getMarkdown()).toBe(value.name);
+
+                    await disconnect();
+                }
+            );
+        }
+    });
+
+    describe('Should return markdown with escape character (back slash) when various special markdown syntax are assigned', () => {
+        const specialMarkdownStrings: { name: string, value: string }[] = [
+            { name: '> blockquote', value: '\\> blockquote' },
+            { name: '`code`', value: '\\`code\\`' },
+            { name: '```fence```', value: '\\`\\`\\`fence\\`\\`\\`' },
+            { name: '~~Strikethrough~~', value: '\\~\\~Strikethrough\\~\\~' },
+            { name: '# Heading 1', value: '\\# Heading 1' },
+            { name: '## Heading 2', value: '\\## Heading 2' },
+            { name: '### Heading 3', value: '\\### Heading 3' },
+            { name: '[link](url)', value: '\\[link\\](url)' },
+            {
+                name: '[ref][link] [link]:url',
+                value: '\\[ref\\]\\[link\\] \\[link\\]:url'
+            },
+            { name: '![Text](Image)', value: '!\\[Text\\](Image)' },
+            { name: '---', value: '\\---' },
+            { name: '***', value: '\\*\\*\\*' },
+            { name: '___', value: '\\__\\_' },
+            { name: '-Infinity', value: '\\-Infinity' },
+            { name: '-2147483648/-1', value: '\\-2147483648/-1' }
+        ];
+
+        const focused: string[] = [];
+        const disabled: string[] = [];
+        for (const value of specialMarkdownStrings) {
+            const specType = getSpecTypeByNamedList(value, focused, disabled);
+            // eslint-disable-next-line @typescript-eslint/no-loop-func
+            specType(
+                `special markdown string "${value.name}" returns as plain text "${value.value}" with added esacpe character`,
+                // eslint-disable-next-line @typescript-eslint/no-loop-func
+                async () => {
+                    element.setMarkdown(value.name);
+
+                    await connect();
+
+                    expect(element.getMarkdown()).toBe(value.value);
+
+                    await disconnect();
+                }
+            );
+        }
+    });
+
+    describe('Should return markdown without any changes when various wacky string values are assigned', () => {
+        const focused: string[] = [];
+        const disabled: string[] = [];
+
+        const wackyStringsWithoutSpecialMarkdownCharacter = [
```

Comment:
Removed `-Infinity` and `-2147483648/-1` from the wacky strings and asserted it in the previous block as it contains special markdown character in it, Remaining wacky strings were added here in `wackyStringsWithoutSpecialMarkdownCharacter` and asserted.

Reply:
Notice how below a filter was used to exclude some strings from the list. Don't copy and paste and modify the list, prefer to exclude items from the list.

Reply:
And then make a separate test where those excluded items are explicitly tested so someone can clearly see how just that set of strings are tested. (currently they are merged up into the above unrelated special markdown strings test)

Reply:
Updated as per your suggestion.